gems/sorcery/0.16/sorcery.rbs

```diff
@@ -0,0 +1,22 @@
+module Sorcery
+  module Model
+    def authenticates_with_sorcery!: () -> void
+
+    module ClassMethods[T]
+      def authenticate: (*String credentials) ?{ () -> untyped } -> (T | nil)
+    end
+  end
+
+  module Controller
+    module InstanceMethods[T]
+      def require_login: () -> (nil | void)
+      def login: (*String credentials) ?{ (T, nil) -> untyped } -> (T | nil | untyped)
```

Comment:
Thank you for your advice.
I was missing the `yield` call. Line 62 in #login method always receive nil as the second argument.
`failure_reason` returns symbol (:invalid_login, :invalid_password or :inactive) or untyped. In this case, what is appropriate types?
I tried to think type below. How about do you think?

```ruby
module Controller
    module InstanceMethods[T]
      ...
      def login: (*String credentials) ?{ (T, (:invalid_login | :invalid_password | :inactive | nil | untyped)) -> untyped } -> (T | nil | untyped)
```

ref: https://github.com/Sorcery/sorcery/blob/4485701dff638bfbca428bec94dcf4de5215dd29/lib/sorcery/controller.rb#L62

Reply:
Defining alias type is great idea!
I defined alias type as failure_reason because it can be used for login and authenticate methods. 
d6ea5a0dd6671cb15acaccb66311f52ce80ab7ad 1eac1244fbf74c6a83f7e519eca9ff8dcbc8e2b8